server/src/main/java/org/elasticsearch/search/vectors/KnnSearchBuilder.java

```diff
@@ -459,6 +485,7 @@ public void writeTo(StreamOutput out) throws IOException {
         }
         out.writeNamedWriteableCollection(filterQueries);
         out.writeFloat(boost);
+        out.writeOptionalString(queryName);
```

Comment:
The reading and writing to the transport stream needs to be protected and backwards compatible. 

There are examples of this elsewhere, but a new version needs added to Transport versions and you should check the stream version before reading and writing.

Reply:
Just pushed a commit that adds version checks for the transport stream and updates the Transport versions. Commit hash is [[7454e8b](https://github.com/elastic/elasticsearch/pull/107645/commits/7454e8b166baec952e8e2a6829d73c3546bdf1f7)]. Could you take another look and let me know if it's all good now?